apps/rewards/src/index.ts

```diff
@@ -0,0 +1,223 @@
+import {
+  TreasuryManager__factory,
+  ERC20__factory,
+  Curve2TokenConvexVaultABI,
+  Curve2TokenConvexVault__factory,
+  Curve2TokenConvexVault,
+} from '@notional-finance/contracts';
+import {
+  DEX_ID,
+  Network,
+  TRADE_TYPE,
+  getProviderFromNetwork,
+} from '@notional-finance/util';
+import { BigNumber, ethers } from 'ethers';
+
+export interface Env {
+  NETWORK: string;
+  TREASURY_MANAGER_ADDRESS: string;
+  TX_RELAY_URL: string;
+  TX_RELAY_AUTH_TOKEN: string;
+  ZERO_EX_PRICE_URL: string;
+  ZERO_EX_API_KEY: string;
+}
+
+const TRADE_PARAMS = 'tuple(uint16,uint8,uint256,bool,bytes)';
+const SINGLE_SIDED_TRADE_PARAMS = `tuple(address,address,uint256,${TRADE_PARAMS})`;
+const VAULT_TRADE_PARAMS = `tuple(${SINGLE_SIDED_TRADE_PARAMS},${SINGLE_SIDED_TRADE_PARAMS})`;
+const ONE = BigNumber.from('1000000000000000000');
+
+// TODO: fetch these from DB or blockchain
+const vaults = [
```

Comment:
@weitianjie2000 Can you put this in a separate config file? I guess not all vaults will need reward reinvestment so it will be easier to maintain a config file of ones that do